### PR TITLE
milestone 2 changes

### PR DESCRIPTION
**Pull request by sumanvpacewisdom**

### CODE REVIEW

File: src/controllers/v1/reports.js [hunk 1]
```diff
@@ -23,4 +24,34 @@ module.exports = class Reports {
...
+		try {
+			let entity_type = {}
+			const reportData = await reportService.getReportData(
+				req.decodedToken.id,
+				req.query.report_code,
+				req.query.report_role ? req.query.report_role : common.MENTEE_ROLE,
+				req.query.start_date ? req.query.start_date : '',
+				req.query.end_date ? req.query.end_date : '',
+				req.query.session_type ? req.query.session_type : common.ALL,
+				req.query.entity_type,
```
Comment (rakeshSgr) [inline src/controllers/v1/reports.js:47]: @sumanvpacewisdom how ur getting entity value?

File: src/controllers/v1/reports.js [hunk 2]
```diff
@@ -23,4 +24,34 @@ module.exports = class Reports {
...
+				req.query.sort_column ? req.query.sort_column : '',
+				req.query.sort_type ? req.query.sort_type : '',
+				req.query.download_csv ? req.query.download_csv : false
+			)
```
Comment (rakeshSgr) [inline src/controllers/v1/reports.js:51]: @sumanvpacewisdom pagination is not require ?

File: src/database/migrations/20241217110259-update-reports-data.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+	async up(queryInterface, Sequelize) {
+		// Insert the report data into the reports table
+		await queryInterface.bulkInsert('reports', [
+			{
+				code: 'total_number_of_sessions_attended',
+				title: 'Total number of sessions attended',
+				description: 'Total sessions attended by user in big number',
```
Comment (rakeshSgr) [inline src/database/migrations/20241217110259-update-reports-data.js:11]: @sumanvpacewisdom use proper sentence  or keep it blank

File: src/database/migrations/20241217110259-update-reports-data.js [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+				report_type_id: 1,
```
Comment (rakeshSgr) [inline src/database/migrations/20241217110259-update-reports-data.js:12]: @sumanvpacewisdom we should map to report type title i think, title should be primary key in report type

File: src/database/migrations/20241217110259-update-reports-data.js [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+				created_at: Sequelize.literal('CURRENT_TIMESTAMP'),
+				updated_at: Sequelize.literal('CURRENT_TIMESTAMP'),
+				config: '{}',
+				organization_id: 1,
```
Comment (rakeshSgr) [inline src/database/migrations/20241217110259-update-reports-data.js:16]: @sumanvpacewisdom use the default org id here

File: src/database/migrations/20241217110259-update-reports-data.js [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+			},
+			{
+				code: 'mentee_session_details',
+				title: 'Session details',
+				description: 'Mentee session details table with pagination and downloadable pdf',
+				report_type_id: 3,
+				created_at: Sequelize.literal('CURRENT_TIMESTAMP'),
+				updated_at: Sequelize.literal('CURRENT_TIMESTAMP'),
+				config: JSON.stringify({
+					columns: [
```
Comment (rakeshSgr) [inline src/database/migrations/20241217110259-update-reports-data.js:46]: @sumanvpacewisdom please check with sansriti or product team, which the columns should be filterable.
I think vishwa has this list, discuss with him once

File: src/database/models/reportQueries.js [hunk 7]
```diff
@@ -51,7 +50,7 @@ module.exports = (sequelize, DataTypes) => {
 	// Define associations
 	ReportQuery.associate = (models) => {
 		ReportQuery.belongsTo(models.Report, {
-			foreignKey: 'report_id',
+			foreignKey: 'report_code',
```
Comment (rakeshSgr) [inline src/database/models/reportQueries.js:53]: @sumanvpacewisdom if ur reference in the query, we need to check it works in citus or not

File: src/database/queries/reportRoleMapping.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async deleteReportRoleMappingById(id) {
+		try {
+			const deletedRows = await ReportRoleMapping.destroy({
+				where: { id },
+				force: true, // Hard delete
```
Comment (rakeshSgr) [inline src/database/queries/reportRoleMapping.js:49]: @sumanvpacewisdom why we are doing hard delete any reason ?

File: src/services/reports.js [hunk 9]
```diff
@@ -100,4 +106,88 @@ module.exports = class ReportsHelper {
...
+	static async getReportData(
+		userId,
+		reportCode,
+		reportRole,
+		startDate,
+		endDate,
+		sessionType,
+		entityType,
+		sortColumn,
+		sortType
```
Comment (rakeshSgr) [inline src/services/reports.js:127]: @sumanvpacewisdom pagination is required in table right ?

File: src/controllers/v1/reports.js (hunk 1 above)
Comment (sumanvpacewisdom) [reply src/controllers/v1/reports.js:47]: @rakeshSgr they should send the entitytype value directly , beacause in the getfilter api they will get entityType label and value with , actually this is entities value then need to send in the entity_type

File: src/controllers/v1/reports.js (hunk 2 above)
Comment (sumanvpacewisdom) [reply src/controllers/v1/reports.js:51]: @rakeshSgr we need , that includes in 3rd milestone , i will include it

**Pull request merged**